# Сессионное задание по курсу «Математический анализ и теория вероятностей»

## Итоговое распределение по заданиям
Задание 1 — 8 баллов\
Задание 2 — 5 баллов\
Задание 3 — 8 баллов\
Задание 4 — 11 баллов\
Задание 5 — 8 баллов\
Задание 6 — 5 баллов\
Задание 7 — 5 баллов\
Сумма: 50 баллов.

### **Задание 1**
Найдите производные функций. Подробно распишите решение. Задание должно быть решено без использования Python.

**a) $\Large y=e^{x \ln (3 x+1)}$**

Заданная функция представляет собой сложную функцию от одной переменной. Производная сложной функции вычисляется по формуле, представленной ниже.

$\hspace{0.5cm}$a.1) ${e^{u\left(x\right)}}'=e^{u\left(x\right)}\,\cdot\,{u}'\left(x\right) $ - Формула для вычисления сложной функции\
$\hspace{0.5cm}$${e^{u\left(x\right)}}'=e^{u\left(x\right)}\,\cdot\,{u}'\left(x\right) = {e}^{x\,\ln\left(3\,x+1\right)}\cdot \left(x\,\ln\left(3\,x+1\right)\right)'$

$\hspace{0.5cm}$a.2) Первый множитель представляет собой заданную функцию.

$\hspace{0.5cm}$$e^{u\left(x\right)} = e^{x \ln (3 x+1)}$

$\hspace{0.5cm}$a.3)Второй множитель представляе собой производную от произведения переменной $x$ с логарифмом. Производная от произведения вычисляется по следующей формуле.

$\hspace{0.5cm}$$(u\cdot v)'=u'\cdot v\;+\,\,v'\cdot u$ - Формула для вычисления производной от произведения.

$\hspace{0.5cm}$${u}'\left(x\right) = (x \ln (3 x+1))' =
(u\cdot v)'=u'\cdot v\;+\,\,v'\cdot u =
\left(\left(x\right)'\cdot \ln\left(3\,x+1\right)+\left(\ln\left(3\,x+1\right)\right)'\cdot x\right) =  
\left(1\cdot \ln\left(3\,x+1\right)+\dfrac{1}{3\,x+1}\cdot 3\cdot x\right)$

$\hspace{1cm}$a.3.1) Производная от $x$ равна единице $(x)' = 1$.

$\hspace{1cm}$Данное выражение также представляет собой сложную функцию, поэтому его производная вычисляется по формуле, упомянутой выше.

$\hspace{1cm}\left(\ln\left(3\,x+1\right)\right)' = \dfrac{1}{3\,x+1}\cdot \left(3\,x+1\right)'= \dfrac{1}{3\,x+1}\cdot 3$

$\hspace{1cm}$Производная от данного выражения равна трем.

$\hspace{1cm}$$\left(3\,x+1\right)' = 3$

Ответ:$\hspace{0.5cm}$ Производная функции $y=e^{x \ln (3 x+1)}$ равна

${e}^{x\,\ln\left(3\,x+1\right)}\,\left(\ln\left(3\,x+1\right)+\dfrac{3\,x}{3\,x+1}\right)$
\
\
\
\



б) $\Large y=e^{x^{2}} \sqrt{x^{3}+4 x^{2}-7}$

Заданная функция представляет собой произведение сложных функций от одной переменной. Производная от произведения вычисляется по следующей формуле.

$\hspace{0.5cm}$$(u\cdot v)'=
u'\cdot v\;+\,\,v'\cdot u$ - Формула для вычисления производной от произведения.

$\hspace{0.5cm}$$u'\cdot v\; =
\left({e}^{{x}^{2}}\right)'\cdot \sqrt{{x}^{3}+4\,{x}^{2}-7} =
{e}^{{x}^{2}}\cdot 2 \cdot x \cdot \sqrt{{x}^{3}+4\,{x}^{2}-7} =
2 x {e}^{{x}^{2}}\sqrt{{x}^{3}+4\,{x}^{2}-7}$

$\hspace{0.5cm}$$\,\,v'\cdot u =
\left(\sqrt{{x}^{3}+4\,{x}^{2}-7}\right)'\cdot {e}^{{x}^{2}} =
{\dfrac{1}{2}\cdot \dfrac{1}{\sqrt{{x}^{3}+4\,{x}^{2}-7}} \cdot ({3 x^2 + 8 x}}) \cdot {e}^{{x}^{2}} =
\dfrac{\left(3\,{x}^{2}+8\,x\right)\,{e}^{{x}^{2}}}{2\,\sqrt{{x}^{3}+4\,{x}^{2}-7}}$

$\hspace{0.5cm}$$(u\cdot v)'=
u'\cdot v\;+\,\,v'\cdot u =
2 x {e}^{{x}^{2}}\sqrt{{x}^{3}+4\,{x}^{2}-7} \hspace{0.5cm}+\hspace{0.5cm}\dfrac{\left(3\,{x}^{2}+8\,x\right)\,{e}^{{x}^{2}}}{2\,\sqrt{{x}^{3}+4\,{x}^{2}-7}} $

Далее необходимо вычислить производные сложных функций. Сложные функции вычисляются по следующим формулам.

$\hspace{0.5cm}$ ${e^{u\left(x\right)}}'=
e^{u\left(x\right)}\,\cdot\,{u}'\left(x\right) $ - Формула для вычисления производной сложной функции от $e$

$\hspace{0.5cm}$ ${e^{u\left(x\right)}}'=
\left({e}^{{x}^{2}}\right)' =
{e}^{{x}^{2}}\cdot \left({x}^{2}\right)' =
{e}^{{x}^{2}}\cdot 2 \cdot x$

$\hspace{0.5cm}$ $\left({x}^{2}\right)' =
2 \cdot x$

$\hspace{0.5cm}$ ${{{u}^{\mathrm{n}}\left(x\right)}}'=
\mathrm{n}\cdot {u}^{\mathrm{n}-1}\left(x\right)\,\cdot\,{u}'\left(x\right)$ - Формула для вычисления производной сложной функции от подкоренного выражения

$\hspace{0.5cm}$ ${{{u}^{\mathrm{n}}\left(x\right)}}' =
\left(\sqrt{{x}^{3}+4\,{x}^{2}-7}\right)' =
\left(({{x}^{3}+4\,{x}^{2}-7})^{\frac{1}{2}}\right)' =
\dfrac{1}{2}\cdot \dfrac{1}{\sqrt{{x}^{3}+4\,{x}^{2}-7}}\cdot \left({x}^{3}+4\,{x}^{2}-7\right)' =
\dfrac{1}{2}\cdot \dfrac{1}{\sqrt{{x}^{3}+4\,{x}^{2}-7}} \cdot ({3 x^2 + 8 x})$

Вычислим производную от линейной функции. Производная суммы равна сумме производных.

$\hspace{0.5cm}$ $\left({x}^{3}+4\,{x}^{2}-7\right)' =
\left(\left({x}^{3}\right)'+4\cdot \left({x}^{2}\right)'-\left(7\right)'\right) =
\left(3\cdot {x}^{2}+4\cdot 2\cdot x-0\right) =
3 x^2 + 8 x$

$\hspace{0.5cm} \left({x}^{3}\right)' =
3 \cdot x^2$

$\hspace{0.5cm} 4\cdot \left({x}^{2}\right)' =
4 \cdot 2 \cdot x$

$\hspace{0.5cm} \left(7\right)' =
0$



Ответ:$\hspace{0.5cm}$ Производная функции $y=e^{x^{2}} \sqrt{x^{3}+4 x^{2}-7}$ равна

$2 x {e}^{{x}^{2}}\sqrt{{x}^{3}+4\,{x}^{2}-7} \hspace{0.5cm}+\hspace{0.5cm}\dfrac{\left(3\,{x}^{2}+8\,x\right)\,{e}^{{x}^{2}}}{2\,\sqrt{{x}^{3}+4\,{x}^{2}-7}}$
\
\
\

в) $\Large y=\frac{x \sqrt{1-x^{2}}}{1+x^{2}}$

Функция представляет собой частное двух других функций. Производна от частного вычисляется по следующей формуле.

$\hspace{0.5cm}$ $\left(\frac{u}{v}\right)'=
\frac{u'\cdot v\,-\,v'\cdot u}{v^{2}} =
\frac{\left(x\,\sqrt{1-{x}^{2}}\right)'\cdot \left({x}^{2}+1\right)-\left({x}^{2}+1\right)'\cdot x\,\sqrt{1-{x}^{2}}}{\left({{x}^{2}+1}\right)^{2}} =
\frac{\bigg(\large\sqrt{1-{x}^{2}}-\dfrac{x^2}{\sqrt{1-{x}^{2}}}\bigg) \cdot (x^2+1) - 2 x^2 \cdot \sqrt{1-{x}^{2}}}{\large\left({{x}^{2}+1}\right)^{2}}$

$\hspace{0.5cm}$${u} =
{x \sqrt{1-x^{2}}}$

$\hspace{0.5cm}$${v} =
{1+x^{2}}$

Вычислим знаменатель.

$\hspace{0.5cm}$${\left(x\,\sqrt{1-{x}^{2}}\right)'\cdot \left({x}^{2}+1\right)-\left({x}^{2}+1\right)'\cdot x\,\sqrt{1-{x}^{2}}} = \bigg(\sqrt{1-{x}^{2}}-\dfrac{2 x^2}{2\sqrt{1-{x}^{2}}}\bigg) \cdot (x^2+1) - 2 x^2 \cdot \sqrt{1-{x}^{2}}
$

$\hspace{0.5cm}$${\left(x\,\sqrt{1-{x}^{2}}\right)}' - $
Данная производная, производная от произведения. Производная от произведения вычисляется по следующей формуле.

$\hspace{0.5cm}$$(u\cdot v)'=
u'\cdot v\;+\,\,v'\cdot u$ - Формула для вычисления производной от произведения.

$\hspace{0.5cm}$$(u\cdot v)'=
u'\cdot v\;+\,\,v'\cdot u = {\left(x\,\sqrt{1-{x}^{2}}\right)}' = {\left(x\right)'\cdot \sqrt{1-{x}^{2}}+\left(\sqrt{1-{x}^{2}}\right)'\cdot x} = 1 \cdot \sqrt{1-{x}^{2}}-\dfrac{2 x^2}{2\sqrt{1-{x}^{2}}}$

$\left(\sqrt{1-{x}^{2}}\right)'$ - производная сложной функции. Сложные функции вычисляются по следующим формулам.

$\hspace{0.5cm}$ ${{{u}^{\mathrm{n}}\left(x\right)}}'=
\mathrm{n}\cdot {u}^{\mathrm{n}-1}\left(x\right)\,\cdot\,{u}'\left(x\right)$ - Формула для вычисления производной сложной функции от подкоренного выражения

$\hspace{0.5cm}$ ${{{u}^{\mathrm{n}}\left(x\right)}}'= \left(\sqrt{1-{x}^{2}}\right)' = (({1-{x}^{2}})^{\frac{1}{2}})' = \dfrac{1}{2}\cdot \dfrac{1}{\sqrt{1-{x}^{2}}}\cdot \left(1-{x}^{2}\right)' = -\dfrac{2 x}{2\sqrt{1-{x}^{2}}}$

Производная разности равна разности производных.

$\hspace{0.5cm}$ $\left(1-{x}^{2}\right)' = (1)' - (x^2)' = 0 - 2 \cdot x = - 2 x$

Производная суммы равна сумме производных.

$\hspace{0.5cm}$$\left({x}^{2}+1\right)' = (x^2)' + (1)' = 2 x$

Ответ:$\hspace{0.5cm}$ Производная функции $y=\frac{x \sqrt{1-x^{2}}}{1+x^{2}}$ равна

$\frac{\bigg(\large{\sqrt{1-{x}^{2}}}-\dfrac{x^2}{\sqrt{1-{x}^{2}}}\bigg) \cdot (x^2+1) - 2 x^2 \cdot \sqrt{1-{x}^{2}}}{\large\left({{x}^{2}+1}\right)^{2}}$



### **Задание 2**
Найдите безусловные экстремумы функции:

$u=3 x^{3}+y^{2}+z^{2}+6 x y-2 z+1$

#### **1.1. Вычисление частных производных**
В задаче представлена функция нескольких переменных $u(x, y, z)=3 x^{3}+y^{2}+z^{2}+6 x y-2 z+1$. Производные для функций нескольких переменных называются частными производными.
$$
\begin{align}
\text{Частная производная первого порядка f от x: }\hspace{1cm} f'_x &= 3 \cdot3 \cdot x^2 + 6 y
\end{align}
$$
$$
\begin{align}
\text{Частная производная первого порядка f от y: }\hspace{1cm} f'_y &= 2 \cdot y + 6 x
\end{align}
$$
$$
\begin{align}
\text{Частная производная первого порядка f от z: }\hspace{1cm} f'_z &= 2 z - 2
\end{align}
$$

#### **1.2. Составление системы уравнений**
Далее необходимо найти точки, в которых частные производные принимают нулевые значения. Для этого необходимо приравнять частные производные к нулю и составить систему уравнений из полученных равенств.

Система уравнений частных производных, приравненных к нулю:
$$
 \begin{equation}
    \begin{cases}
        9 x^2 + 6 y = 0\\
        2 y + 6 x = 0 \\
        2 z - 2 = 0
    \end{cases}
\end{equation}
$$

#### **1.3. Решение системы уравнений**
Полученную систему уравнений будем решать методом подстановки.

Из третьего уравнения выразим $z$, из второго уравнения выразим $y$, подставим получившееся значение в первое уравнение, также первое уравнение разделим на 3:
$$
 \begin{equation}
 \left.
    \begin{cases}
        9 x^2 + 6 y = 0\\
        2 y + 6 x = 0 \\
        2 z - 2 = 0
    \end{cases}
    \right. \Rightarrow
    \left.
    \begin{cases}
        9 x^2 + 6 (-3 x) = 0 | :3\\
        y = -3 x \\
        z = 1
    \end{cases}
    \right. \Rightarrow
     \left.
    \begin{cases}
        3 x^2 - 6 x = 0\\
        y = - 3 x \\
        z = 1
    \end{cases}
    \right.
\end{equation}
$$

Для нахождения значения $x$ необходимо решить квадратное уравнение $3 x^2 - 6 x = 0$, для этого воспользуемся теоремой Виета. Теорема Виета устанавливает связь между корнями квадратного уравнения $ax^{2}+bx+c=0$ и его коэффициентами: сумма корней $x_{1}+x_{2}$ равна $-b/a$, а произведение корней $x_{1}\cdot x_{2}$ равно $c/a$.
Для нашего случая:
1) Cумма корней $\hspace{0.5cm}x_{1}+x_{2} = -b/a = -(-6/3) = 2$

2) Произведение корней $\hspace{0.5cm}x_{1}\cdot x_{2} = c/a = 0/3 = 0$

Далее методом подбора находим значения корней $x_{1}$, $x_{2}$:
1) $x_{1} = 2$ \
Проверим, подставив значение в уравнение: $\hspace{0.5cm}3 x^2 - 6 x = 0 \hspace{0.5cm}\Rightarrow\hspace{0.5cm} 3 \cdot 2^2 - 6 \cdot 2 = 0 \hspace{0.5cm}\Rightarrow\hspace{0.5cm} 12 - 12 = 0 \hspace{0.5cm}\Rightarrow\hspace{0.5cm} 0 = 0$

2) $x_{2} = 0$ \
Проверим, подставив значение в уравнение: $\hspace{0.5cm}3 x^2 - 6 x = 0 \hspace{0.5cm}\Rightarrow\hspace{0.5cm} 3 \cdot 0^2 - 6 \cdot 0 = 0  \hspace{0.5cm}\Rightarrow\hspace{0.5cm} 0 = 0$

Далее найдем значения остальных переменных, подставив значения $x_{1}$, $x_{2}$ в систему уравнений:

1) Для $x_{1} = 2$

$$
 \begin{equation}
     \left.
    \begin{cases}
        3 \cdot 2^2 - 6 \cdot 2 = 0\\
        y = - 3 \cdot 2 \\
        z = 1
    \end{cases}
    \right. \Rightarrow
    \begin{cases}
        0 = 0\\
        y = -6 \\
        z = 1
    \end{cases}
\end{equation}
$$

$$
x_{1} = 2;\hspace{0.5cm} y_{1} = -6;\hspace{0.5cm} z_{1} = 1
$$

2) Для $x_{1} = 0$

$$
 \begin{equation}
     \left.
    \begin{cases}
        3 \cdot 0^2 - 6 \cdot 0 = 0\\
        y = - 3 \cdot 0 \\
        z = 1
    \end{cases}
    \right. \Rightarrow
    \begin{cases}
        0 = 0\\
        y = 0 \\
        z = 1
    \end{cases}
\end{equation}
$$

$$
x_{1} = 0;\hspace{0.5cm} y_{1} = 0;\hspace{0.5cm} z_{1} = 1
$$
#### **1.4. Анализ второго порядка**
Необходимое условие экстремума функции: если дифференцируемая функция $u(x, y, z)$ имеет экстремум в некоей точке, то все три производные первого порядка в данной точке равны нулю. Такую точку называют стационарной. В нашем случае стационарных точек две, назовем их $M_1$ и $M_2$: $M_1(2;-6;1)$ и $M_2(0;0;1)$.
Далее для проверки наличия экстремума необходимо найти все частные производные второго порядка, вычислить их в точках $M_1$ и $M_2$, составить матрицу Гессе.
Найдем все частные производные второго порядка:

$$
\begin{align}
\text{Частная производная первого порядка f от x: }\hspace{1cm} f'_x &= 3 \cdot3 \cdot x^2 + 6 y
\end{align}
$$
$$
\begin{align}
\text{Частная производная первого порядка f от y: }\hspace{1cm} f'_y &= 2 \cdot y + 6 x
\end{align}
$$
$$
\begin{align}
\text{Частная производная первого порядка f от z: }\hspace{1cm} f'_z &= 2 z - 2
\end{align}
$$

$$
\begin{align}
\text{Частная производная второго порядка $f'_x$ по x: }\hspace{1cm} f''_{xx} &= 18 x\newline
\text{Частная производная второго порядка $f'_y$ по y: }\hspace{1cm} f''_{yy} &= 2 \newline
\text{Частная производная второго порядка $f'_z$ по z: }\hspace{1cm} f''_{zz} &= 2 \\
\end{align}
$$

$$
\begin{align}
\text{Смешанная частная производная второго порядка $f'_x$ по y: }\hspace{1cm} f''_{xy} &= 6\newline
\text{Смешанная частная производная второго порядка $f'_x$ по z: }\hspace{1cm} f''_{xz} &= 0\newline
\text{Смешанная частная производная второго порядка $f'_y$ по x: }\hspace{1cm} f''_{yx} &= 6\newline
\text{Смешанная частная производная второго порядка $f'_y$ по z: }\hspace{1cm} f''_{yz} &= 0\newline
\text{Смешанная частная производная второго порядка $f'_z$ по x: }\hspace{1cm} f''_{zx} &= 0\newline
\text{Смешанная частная производная второго порядка $f'_z$ по y: }\hspace{1cm} f''_{zy} &= 0
\end{align}
$$
Далее вычислим значания частных производных второго порядка в стационарных точках. В нашем случае все производные второго порядка, кроме $f''_{xx}$, равны константам, а значит и в стационарных точках они тоже будут равны константам. Вычислим значение $f''_{xx}$ в стационарных точках:

1) В точке $M_1(2;-6;1)$:
$
\hspace{0.5cm}
f''_{xx}(M_1) = 18 * 2 = 36
$

2) В точке $M_2(0;0;1)$:
$
\hspace{0.5cm}
f''_{xx}(M_2) = 12 * 0 = 0
$

Далее составим матрицы Гессе. В нашем случае их две:
$$
Hesse_1 =
\begin{pmatrix}
f''_{xx}(M_1) & f''_{xy} & f''_{xz} \\
f''_{yx} & f''_{yy} & f''_{yz} \\
f''_{zx} & f''_{zy} & f''_{zz}
\end{pmatrix}
=
\begin{pmatrix}
36 & 6 & 0 \\
6 & 2 & 0 \\
0 & 0 & 2
\end{pmatrix}
$$    

$$
Hesse_2 =
\begin{pmatrix}
f''_{xx}(M_2) & f''_{xy} & f''_{xz} \\
f''_{yx} & f''_{yy} & f''_{yz} \\
f''_{zx} & f''_{zy} & f''_{zz}
\end{pmatrix}
=
\begin{pmatrix}
0 & 6 & 0 \\
6 & 2 & 0 \\
0 & 0 & 2
\end{pmatrix}
$$  

Далее нужно вычислить угловые миноры. Это определители, которые "разрастаются" из левого верхнего угла.
1) Для матрицы $Hesse_1$(миноры обозначим $\delta$):
$$
Hesse_1 =
\begin{pmatrix}
36 & 6 & 0 \\
6 & 2 & 0 \\
0 & 0 & 2
\end{pmatrix}
$$

$$
\delta_{11} =
\begin{vmatrix}
36
\end{vmatrix}
= 36
$$

$$
\delta_{12} =
\begin{vmatrix}
36 & 6 \\
6 & 2 \\
\end{vmatrix}
= 36*2 - 6*6 = 72 - 36 = 36
$$

$$
\delta_{13} =
\begin{vmatrix}
36 & 6 & 0 \\
6 & 2 & 0 \\
0 & 0 & 2
\end{vmatrix}
= 36*2*2 + 0*6*0 + 6*0*0 - 0*2*0 - 36*0*0 -6*6*2 = 144 + 0 + 0 - 0 - 0 - 72 = 72
$$

$\delta_{11} = 36 > 0$, $\delta_{12} = 36 > 0$, $\delta_{13} = 72 > 0$, все угловые миноры матрицы $Hesse_1$ положительны, следовательно и сама матрица положительно определена в точке $M_1(2;-6;1)$, а это значит, что для функции $u=3 x^{3}+y^{2}+z^{2}+6 x y-2 z+1$ в точке $M_1$ находится минимум.

2) Для матрицы $Hesse_2$(миноры обозначим $\delta$):
$$
Hesse_2 =
\begin{pmatrix}
0 & 6 & 0 \\
6 & 2 & 0 \\
0 & 0 & 2
\end{pmatrix}
$$

$$
\delta_{21} =
\begin{vmatrix}
0
\end{vmatrix}
= 0
$$

$$
\delta_{22} =
\begin{vmatrix}
0 & 6 \\
6 & 2 \\
\end{vmatrix}
= 0*2 - 6*6 = 0 - 36 = - 36
$$

$$
\delta_{23} =
\begin{vmatrix}
0 & 6 & 0 \\
6 & 2 & 0 \\
0 & 0 & 2
\end{vmatrix}
= 0*2*2 + 0*6*0 + 6*0*0 - 0*2*0 - 0*0*0 -6*6*2 = 0 + 0 + 0 - 0 - 0 - 72 = - 72
$$

$\delta_{21} = 0$, $\delta_{22} = - 36 < 0$, $\delta_{23} = - 72 < 0$, один угловой минор матрицы $Hesse_2$ равен нулю, остальные отрицательны. Для того, чтобы матрица была положительно определённой, необходимо, чтобы все её угловые миноры, либо все собственные значения были положительными. А для того, чтобы матрица была отрицательно определённой, необходимо, чтобы знаки угловых миноров чередовались, начиная с отрицательного, либо чтобы все собственные значения были отрицательными. В случае матрицы $Hesse_2$ не выполняются ни первое, ни второе условие, а значит, что для функции $u=3 x^{3}+y^{2}+z^{2}+6 x y-2 z+1$ точка $M_2(0;0;1)$ является седловой, то есть точкой, в которой функция не имеет ни локального минимума, ни локального максимума.

#### **1.5. Запись результата и выводов**
По условиям задачи необходимо было найти экстремум функции $\hspace{0.5cm}u=3 x^{3}+y^{2}+z^{2}+6 x y-2 z+1$.

Для поиска экстремума необходимо было найти стационарные точки, в которых возможен экстремум. А также классифицировать стационарные точки, чтобы определить существует ли экстремумы в данных точках.

В процессе решения были найдены частные производные первого порядка, также были найдены стационарные точки, в которых все производные первого порядка равны нулю $M_1(2;-6;1)$ и $M_2(0;0;1)$.

Далее были найдены частные производные второго порядка, а также их значения в стационарных точках. Далее из частных производных второго порядка были составлены матрицы Гессе, в результате анализа которых было выяснено, что для функции $u=3 x^{3}+y^{2}+z^{2}+6 x y-2 z+1$ в точке $M_1$ находится минимум, так как матрица Гессе в этой точке положительно определена, а точка $M_2(0;0;1)$ для функции $u=3 x^{3}+y^{2}+z^{2}+6 x y-2 z+1$ является седловой, то есть точкой, в которой функция не имеет ни локального минимума, ни локального максимума.

**Условия задачи выполнены, экстремум найден.**

### **Задание 3**
Найдите условный экстремум функции:

$\large z=e^{-x-y}$

Если задано следующее условие:

$\large \frac{x^{2}}{4}+y^{2}=5$

#### **1) Составление функции множителей Лагранжа**

Нам необходимо найти экстремумы функции $z(x, y)=e^{-x-y}$ при условии $\frac{x^{2}}{4}+y^{2}=5$. Находить экстремумы будем с помощью метода множителей Лагранжа. Для этого необходимо составить функцию Лагранжа, то есть объединить целевую функцию и функцию ограничений внутри одного выражения.

$$
 L(x, y, \lambda) = e^{-x-y} + \lambda(\frac{x^{2}}{4}+y^{2} - 5)
$$

#### **2) Вычисление частных производных и составление системы уравнений**

Далее порядок решения задачи такой же, как и при поиске безусловного экстремума. Необходимо вычислить частные производные первого порядка функции Лагранжа, приравнять их к нулю, составить из них систему уравнений и решить ее. Решения системы уравнений будут координатами стационарных точек.

#### 2.1) Найдем частные производные первого порядка:

Производная от $e^{-x-y}$ это производная от сложной функции. Производная сложной функции вычисляется по формуле, представленной ниже.

$\hspace{0.5cm}$${e_x^{u\left(x\right)}}'=e^{u\left(x\right)}\,\cdot\,{u}'\left(x\right) $ - Формула для вычисления сложной функции

$\hspace{0.5cm}$${e_x^{u\left(x\right)}}'=e^{u\left(x\right)}\,\cdot\,{u}'\left(x\right) = {e}^{-x-y}\cdot \left(-x-y\right)'_x = {e}^{-x-y} \cdot (-1) = -e^{-x-y}$

$\hspace{0.5cm}$${e_y^{u\left(x\right)}}'=e^{u\left(x\right)}\,\cdot\,{u}'\left(x\right) = {e}^{-x-y}\cdot \left(-x-y\right)'_y = {e}^{-x-y} \cdot (-1) = -e^{-x-y}$

Производная разности равна разности производных.

$(-x -y)'_x = (-x)'_x + (-y)'_x = -1 + 0 = -1$

$(-x -y)'_y = (-x)'_y + (-y)'_y = 0 + (-1) = -1$

$$
\begin{align}
\text{Частная производная первого порядка L от x: }\hspace{1cm} L'_x &= -e^{-x-y} + \frac {1}{4}\cdot \lambda \cdot 2 x\newline
\text{Частная производная первого порядка L от y: }\hspace{1cm} L'_y &= -e^{-x-y} + \lambda \cdot 2 y\newline
\text{Частная производная первого порядка L от $\lambda$: }\hspace{1cm} L'_\lambda &= \frac{x^{2}}{4}+y^{2} - 5
\end{align}
$$

#### 2.1) Составим систему уравнений частных производных первого порядка, приравненных к нулю:

$$
 \begin{equation}
    \begin{cases}
        -e^{-x-y} + \frac {1}{2} \lambda x = 0\\
        -e^{-x-y} + 2\lambda y= 0 \\
        \frac{1}{4}x^2+y^{2} - 5 = 0
    \end{cases}
\end{equation}
$$

#### **3) Решение системы**
Полученную систему уравнений будем решать методом подстановки.

Для начала, первое уравнение домножим на $-1$, затем сложим первое и второе уравнение, выразим из второго уравнения $y$ и подставим его в третье уравнение:
$$
 \begin{equation}
    \left.
    \begin{cases}
        -e^{-x-y} + \frac {1}{2} \lambda x = 0|*(-1)\\
        -e^{-x-y} + 2\lambda y= 0 \\
        \frac{1}{4}x^2+y^{2} - 5 = 0
    \end{cases}
    \right.\Rightarrow
    \left.
    \begin{cases}
        e^{-x-y} - \frac {1}{2} \lambda x = 0\\
        -e^{-x-y} + 2\lambda y= 0 \\
        \frac{1}{4}x^2+y^{2} - 5 = 0
    \end{cases}
    \right|(2)+(1) \Rightarrow
    \left.
    \begin{cases}
        e^{-x-y} - \frac {1}{2} \lambda x = 0\\
        2\lambda y - \frac {1}{2} \lambda x= 0 \\
        \frac{1}{4}x^2+y^{2} - 5 = 0
    \end{cases}
    \right. \Rightarrow
    \left.
    \begin{cases}
        e^{-x-y} - \frac {1}{2} \lambda x = 0\\
        y = \frac {x}{4} \\
        \frac{1}{4}x^2+(\frac {x}{4})^{2} - 5 = 0
    \end{cases}
    \right.
\end{equation}
$$

Для нахождения значения $x$ необходимо решить квадратное уравнение $\frac{1}{4}x^2+(\frac {x}{4})^{2} - 5 = 0$, раскроем скобки и упростим его:
$$
\frac{1}{4}x^2+(\frac {x}{4})^{2} - 5 = 0 \hspace{0.5cm}\Rightarrow\hspace{0.5cm}\frac{1}{4}x^2+\frac {x^2}{16} - 5 = 0 |*16\hspace{0.5cm}\Rightarrow\hspace{0.5cm}4x^2+x^2 - 80 = 0 \hspace{0.5cm}\Rightarrow\hspace{0.5cm} 5x^2 = 80 \hspace{0.5cm}\Rightarrow\hspace{0.5cm} x^2 = 16 \hspace{0.5cm}\Rightarrow\hspace{0.5cm} x_1 = 4 \hspace{0.5cm} x_2 = -4$$

Далее найдем значения остальных переменных, подставив значения $x_1$, $x_2$ в систему уравнений:

1) Для $x_1 = 4$

$$
 \begin{equation}
    \left.
        \begin{cases}
        e^{-x-y} - \frac {1}{2} \lambda x = 0\\
        y = \frac {4}{4} \\
        \frac{1}{4}4^2+(\frac {4}{4})^{2} - 5 = 0
    \end{cases}
    \right. \Rightarrow
    \left.
    \begin{cases}
        e^{-4-1} - \frac {1}{2} \lambda 4 = 0\\
        y = 1 \\
        0 = 0
    \end{cases}
    \right. \Rightarrow
    \left.
    \begin{cases}
        \lambda  = \large\frac{e^{-5}}{2}\\
        y = 1 \\
        0 = 0
    \end{cases}
    \right.
\end{equation}
$$

$$
x_{1} = 4;\hspace{0.5cm} y_{1} = 1;\hspace{0.5cm} \lambda_1 = \large\frac{e^{-5}}{2}
$$
2) Для $x_2 = -4$

$$
 \begin{equation}
    \left.
        \begin{cases}
        e^{-x-y} - \frac {1}{2} \lambda x = 0\\
        y = \frac {-4}{4} \\
        \frac{1}{4}(-4)^2+(\frac {-4}{4})^{2} - 5 = 0
    \end{cases}
    \right. \Rightarrow
    \left.
    \begin{cases}
        e^{-(-4)-(-1)} - \frac {1}{2} \lambda (-4) = 0\\
        y = -1 \\
        0 = 0
    \end{cases}
    \right. \Rightarrow
    \left.
    \begin{cases}
        \lambda  = -\large\frac{e^{5}}{2}\\
        y = 1 \\
        0 = 0
    \end{cases}
    \right.
\end{equation}
$$

$$
x_{2} = -4;\hspace{0.5cm} y_{2} = -1;\hspace{0.5cm} \lambda_2 = -\large\frac{e^{5}}{2}
$$

#### **4) Анализ второго порядка**
Необходимое условие экстремума функции: если дифференцируемая функция $\large z(x,y)=e^{-x-y}$ имеет экстремум в некоей точке, при условии $\large \frac{x^{2}}{4}+y^{2}=5$, то все частные производные первого порядка в данной точке равны нулю. Такую точку называют стационарной. В нашем случае стационарных точек две, назовем их $M_1$ и $M_2$: $M_1(4,1)$ и $M_2(-4,-1)$.
Далее для проверки наличия экстремума необходимо найти все частные производные второго порядка, вычислить их в точках $M_1$ и $M_2$, найти частные производные первого порядка от функции ограничений, составить окаймленную матрицу Гессе.
Найдем все частные производные второго порядка:

$$
\begin{align}
\text{Частная производная первого порядка L от x: }\hspace{1cm} L'_x &= -e^{-x-y} + \frac {1}{4}\cdot \lambda \cdot 2 x\newline
\text{Частная производная первого порядка L от y: }\hspace{1cm} L'_y &= -e^{-x-y} + \lambda \cdot 2 y\newline
\text{Частная производная первого порядка функции ограничений $g$ от $x$: }\hspace{1cm} g'_x &= \frac{1}{4} \cdot 2 \cdot x \newline
\text{Частная производная первого порядка функции ограничений $g$ от $y$: }\hspace{1cm} g'_y &= 2y
\end{align}
$$

$$
\begin{align}
\text{Частная производная второго порядка $L'_x$ по x: }\hspace{1cm} L''_{xx} &= e^{-x-y} + \frac {1}{4}\cdot \lambda \cdot 2\\
\text{Частная производная второго порядка $L'_y$ по y: }\hspace{1cm} L''_{yy} &= e^{-x-y} + \lambda \cdot 2\\
\end{align}
$$

$$
\begin{align}
\text{Смешанная частная производная второго порядка $L'_x$ по y: }\hspace{1cm} L''_{xy} &= e^{-x-y}\\
\text{Смешанная частная производная второго порядка $L'_y$ по x: }\hspace{1cm} L''_{yx} &= e^{-x-y}\\
\end{align}
$$

Далее составим окаймленную матрицу Гессе.
$$
Hesse =
\begin{pmatrix}
0 & g'_x & g'_y \\
g'_x & L''_{xx} & L''_{xy} \\
g'_y & L''_{yx} & L''_{yy}
\end{pmatrix}
=
\begin{pmatrix}
0 & \frac{x}{2}  & 2y \\
\frac{x}{2} & (e^{-x-y} + \frac {\lambda}{2}) & e^{-x-y} \\
2y & e^{-x-y} & (e^{-x-y} + 2\lambda)
\end{pmatrix}
$$    

Далее подставим в матрицу Гессе значения переменных в стационарных точках.

1. Для $M_1(4;1)$ и $\lambda_1 = \large\frac{e^{-5}}{2}$
$$
Hesse_1 =
\begin{pmatrix}
0 & \frac{x}{2}  & 2y \\
\frac{x}{2} & (e^{-x-y} + \frac {\lambda}{2}) & e^{-x-y} \\
2y & e^{-x-y} & (e^{-x-y} + 2\lambda)
\end{pmatrix}
=
\begin{pmatrix}
0 & 2 & 2 \\
2 & \frac{5e^{-5}}{4} & e^{-5} \\
2 & e^{-5} & 2e^{-5}
\end{pmatrix}
$$    

2. Для $M_2(-4;-1)$ и $\lambda_2 = -\large\frac{e^{5}}{2}$
$$
Hesse_2 =
\begin{pmatrix}
0 & \frac{x}{2}  & 2y \\
\frac{x}{2} & (e^{-x-y} + \frac {\lambda}{2}) & e^{-x-y} \\
2y & e^{-x-y} & (e^{-x-y} + 2\lambda)
\end{pmatrix}
=
\begin{pmatrix}
0 & -2 & -2 \\
-2 & \frac{5e^{5}}{4} & e^{5} \\
-2 & e^{5} & 0
\end{pmatrix}
$$  

Далее нужно вычислить определители окаймленных гессианов.
1) Для матрицы $Hesse_1$:
$$
|Hesse_1| =
\begin{vmatrix}
0 & 2 & 2 \\
2 & \frac{5e^{-5}}{4} & e^{-5} \\
2 & e^{-5} & 2e^{-5}
\end{vmatrix}
= 0*\frac{5e^{-5}}{4}*2e^{-5} + 2*2*e^{-5} + 2*2*e{-5} - 2*\frac{5e^{-5}}{4}*2 - 0*e^{-5}*e^{-5} - 2*2*2e^{-5} = 0 + 4e^{-5} + 4e^{-5} - 4*\frac{5e^{-5}}{4} - 0  - 8e^{-5} = -{5e^{-5}} \approx -0.03368984830307695, \text{Таким образом в точке $M_1(4;1)$ функция имееет минимум.}
$$    

1) Для матрицы $Hesse_2$:
$$
|Hesse_2| =
\begin{vmatrix}
0 & -2 & -2 \\
-2 & \frac{5e^{5}}{4} & e^{5} \\
-2 & e^{5} & 0
\end{vmatrix}
= 0*\frac{5e^{5}}{4}*0 + (-2)*(-2)*e^5 + (-2)*e^5*(-2) - (-2)*\frac{5e^{5}}{4}*(-2) - 0*e^5*e^5 - (-2)*(-2)*0 = 0 + 4e^5 + 4e^5 - 4\frac{5e^{5}}{4} - 0 - 0 = 7e^5 \approx 1038.888619655892, \text{Таким образом в точке $M_2(-4,-1)$ функция имееет максимум.}
$$   

#### **5) Формулирование итоговых выводов**

По условиям задачи необходимо было найти экстремум целевой функции $\hspace{0.5cm} z(x, y)=e^{-x-y}$ при условии  $\frac{x^{2}}{4}+y^{2}=5
$.

Для поиска экстремума необходимо было составить функцию Лагранжа, найти стационарные точки, найти частные производные второго порядка, составить окаймленную матрицу Гессиана и вычислить ее определитель.

В процессе решения были найдены частные производные первого порядка, была составлена система уравнений частных производных первого порядка, приравненных к нулю. Эта система была решена. Также были найдены стационарные точки, в которых определяется условный экстремум функции. Далее были найдены частные производные второго порядка, составлены окаймленные матрицы Гессиана для стационарных точек и найдены их определители.
Определитель матрицы Гессиана в точке $M_1(4;1), |Hesse_1| = \approx -0.03368984830307695 < 0$, следовательно в этой точке, при условии  $\frac{x^{2}}{4}+y^{2}=5$, находится минимум целевой функции $\hspace{0.5cm} z(x, y)=e^{-x-y}$.

Определитель матрицы Гессиана в точке $M_2(-4;-1), |Hesse_2| = \approx 1038.888619655892 > 0$, следовательно в этой точке, при условии  $\frac{x^{2}}{4}+y^{2}=5$, находится максимум целевой функции $\hspace{0.5cm} z(x, y)=e^{-x-y}$.

**Условия задачи выполнены, условные экстремумы найдены.**

### **Задание 4**
Вам предоставлены данные о людях с наличием или отсутствием заболевания:
Признаки:

Age — возрастная группа.
Test — позитивный или негативный тест на заболевание.
Status — целевая переменная, есть инфекция или нет.
Необходимо реализовать алгоритм Наивного Байеса для решения задачи классификации. Обязательно оцените качество полученного результата по итогу.

In [ ]:
import numpy as np
import pandas as pd

# Загружаем файл
test_data = pd.read_csv('/content/asset-v1_SkillFactory+MIFIML-1sem+2024+type@asset+block@sf_exam (1).csv')
test_data.head()

# Предобработка данных. Удаляем символы, не являющиеся буквами, приводим слова к нижнему регистру.
test_data_clean = test_data.copy()


test_data_clean['Test'] = test_data_clean['Test'].str.replace('\W+', ' ', regex=True)
test_data_clean['Test'] = test_data_clean['Test'].str.replace('\s+', ' ', regex=True).str.strip()
test_data_clean['Test'] = test_data_clean['Test'].str.lower()
test_data_clean['Test'] = test_data_clean['Test'].str.split()

test_data_clean['Age_Group'] = test_data_clean['Age_Group'].str.replace('\W+', ' ', regex=True)
test_data_clean['Age_Group'] = test_data_clean['Age_Group'].str.replace('\s+', ' ', regex=True).str.strip()
test_data_clean['Age_Group'] = test_data_clean['Age_Group'].str.lower()
test_data_clean['Age_Group'] = test_data_clean['Age_Group'].str.split()

test_data_clean['Status'] = test_data_clean['Status'].str.replace('\W+', ' ', regex=True)
test_data_clean['Status'] = test_data_clean['Status'].str.replace('\s+', ' ', regex=True).str.strip()
test_data_clean['Status'] = test_data_clean['Status'].str.lower()

# Объединяем признаки. Разделяем на обучающую и тестовую выборки

test_data_clean['Test_Age_Group'] = test_data_clean['Age_Group'] + test_data_clean['Test']
del test_data_clean['Test']
del test_data_clean['Age_Group']

train_data = test_data_clean.sample(frac=0.8, random_state=42)
test_data = test_data_clean.drop(train_data.index)

train_data = train_data.reset_index(drop=True)
test_data = test_data.reset_index(drop=True)

# Проверяем соотношение значений целевой переменной в обучающей и тестовой выборке. Оно идентично.
print(train_data['Status'].value_counts() / train_data.shape[0] * 100,'\n')
print(test_data['Status'].value_counts() / test_data.shape[0] * 100,'\n')

# Создаем список слов, встречающихся в обучающей выборке.
vocabulary = list(set(train_data['Test_Age_Group'].sum()))
vocabulary

# Для каждого диагноза посчитаем, сколько раз в нем встречается каждое слово.
word_counts_per_sms = pd.DataFrame([
   [row[1].count(word) for word in vocabulary]
   for _, row in train_data.iterrows()], columns=vocabulary)

word_counts_per_sms.head()

# Добавим частоты каждого слова в обучающий датасет.
train_data = pd.concat([train_data, word_counts_per_sms], axis=1)

train_data.head()

# Посчитаем необходимые значения для формулы Байеса.
alpha = 1

Nvoc = len(vocabulary)
Pinf = train_data['Status'].value_counts()['infected'] / train_data.shape[0]
Pnot_inf = train_data['Status'].value_counts()['not_infected'] / train_data.shape[0]
Ninf = train_data.loc[train_data['Status'] == 'infected', 'Test_Age_Group'].apply(len).sum()
Nnot_inf = train_data.loc[train_data['Status'] == 'not_infected', 'Test_Age_Group'].apply(len).sum()

def p_w_inf(word):
    if word in train_data.columns:
        return (train_data.loc[train_data['Status'] == 'infected', word].sum() + alpha) / (Ninf + alpha*Nvoc)
    else:
        return 1

def p_w_not_inf(word):
    if word in train_data.columns:
        return (train_data.loc[train_data['Status'] == 'not_infected', word].sum() + alpha) / (Nnot_inf + alpha*Nvoc)
    else:
        return 1

# Готовим алгоритм классификации

def classify(message):
    p_spam_given_message = Pinf
    p_ham_given_message = Pnot_inf
    for word in message:
        p_spam_given_message *= p_w_inf(word)
        p_ham_given_message *= p_w_not_inf(word)
    if p_ham_given_message > p_spam_given_message:
        return 'not_infected'
    elif p_ham_given_message < p_spam_given_message:
        return 'infected'
    else:
        return 'классификация некорректна'

# Используем тестовые данные
test_data['predicted'] = test_data['Test_Age_Group'].map(classify)
test_data.head()

# Расчитываем метрику качества Accuracy
accuracy = (test_data['predicted'] == test_data['Status']).sum() / test_data.shape[0]
print(f"Правильных предсказаний {accuracy * 100:3f} %",'\n')

test_data.loc[test_data['predicted'] != test_data['Status']].head()

Status
infected        51.351351
not_infected    48.648649
Name: count, dtype: float64 

Status
infected        50.0
not_infected    50.0
Name: count, dtype: float64 

Правильных предсказаний 83.928571 % 



<>:12: SyntaxWarning: invalid escape sequence '\W'
<>:13: SyntaxWarning: invalid escape sequence '\s'
<>:17: SyntaxWarning: invalid escape sequence '\W'
<>:18: SyntaxWarning: invalid escape sequence '\s'
<>:22: SyntaxWarning: invalid escape sequence '\W'
<>:23: SyntaxWarning: invalid escape sequence '\s'
<>:12: SyntaxWarning: invalid escape sequence '\W'
<>:13: SyntaxWarning: invalid escape sequence '\s'
<>:17: SyntaxWarning: invalid escape sequence '\W'
<>:18: SyntaxWarning: invalid escape sequence '\s'
<>:22: SyntaxWarning: invalid escape sequence '\W'
<>:23: SyntaxWarning: invalid escape sequence '\s'
/tmp/ipython-input-209724553.py:12: SyntaxWarning: invalid escape sequence '\W'
  test_data_clean['Test'] = test_data_clean['Test'].str.replace('\W+', ' ', regex=True)
/tmp/ipython-input-209724553.py:13: SyntaxWarning: invalid escape sequence '\s'
  test_data_clean['Test'] = test_data_clean['Test'].str.replace('\s+', ' ', regex=True).str.strip()
/tmp/ipython-input-209724553.py:17: Synt

Status     Test_Age_Group     predicted
0       infected    [old, positive]  not_infected
3   not_infected  [young, positive]      infected
11      infected    [old, positive]  not_infected
12  not_infected  [young, positive]      infected
14      infected    [old, positive]  not_infected

### **Задание 5**
Дана функция:

$f(x)=x^{4}+3 x^{3}-12 x^{2}+7 x-2$

Оптимизируйте её с помощью метода Ньютона, используя язык программирования Python.

За начальную точку возьмите х = 10, необходимую точность 0.0001.

In [ ]:
import sympy
from sympy import *
x = sympy.Symbol("x")

# Запишем данную функцию
expr = x**4 + 3*(x**3) - 12*(x**2) + 7*x -2

# Найдем первую производную
print('Первая производная', expr.diff(x), '\n')

# Найдем вторую производную
print('Вторая производная', expr.diff(x,2), '\n')

# Запишем первую производную
def func1(x):
    return 4*x**3 + 9*x**2 - 24*x + 7

# Запишем вторую производную
def func2(x):
    return 6*(2*x**2 + 3*x - 4)

# Оптимизация функции методом Ньютона
def newtons_method(f, fprime, x0, tol=0.0001):
    iter_count = 0

    # Начальная точка
    x_curr = x0

    # Значение первой производной в начальной точке
    f_val = f(x_curr)
    while (abs(f_val) > tol):
        f_val = f(x_curr)
        f_prime_val = fprime(x_curr)

        # Формула метода Ньютона
        x_curr = x_curr - (f_val)/(f_prime_val)
        iter_count += 1
    return round(x_curr, 4)

newtons_method(f=func1, fprime=func2, x0=10, tol=0.0001)

Первая производная 4*x**3 + 9*x**2 - 24*x + 7 

Вторая производная 6*(2*x**2 + 3*x - 4) 



1.3101

#### **Ответ: Точка минимума функции $f(x)=x^{4}+3 x^{3}-12 x^{2}+7 x-2 \hspace{1cm}  x = 1.3101$**

### **Задание 6**
Дано распределение случайной величины, которая отражает вероятность получения разного количества спам-писем в течение дня:

$$
\begin{array}{|c|c|c|c|c|c|c|}
\hline X \text { (количество спам-писем) } & 0 & 1 & 2 & 3 & 4 & 5 & 6 & 7 \\
\hline \mathrm{P} \text { (вероятность) } & 0.1 & 0.3 & 0.1 & 0.25 & 0.05 & 0.05 & 0.1 & ? \\
\hline
\end{array}
$$

1. Найдите вероятность получения 7 спам-писем.
2. Найдите математическое ожидание для количества полученных писем.
3. Найдите дисперсию для количества полученных писем. Ответ округлите до сотых.

Задание должно быть решено без использования готовых функций Python.

1. Найдем вероятность получения 7 спам-писем.
Сумма вероятностей всех возможных значений случайной величины всегда равна единице. Следовательно:

$P(X = 7) = 1 - P(X = 0) - P(X = 1) - P(X = 2) - P(X = 3) - P(X = 4) - P(X = 5) - P(X = 6) = 1 - 0.1 - 0.3 - 0.1 - 0.25 - 0.05 - 0.05 - 0.1 = 0.05$


$P(X = 7) = 0.05$

$$
\begin{array}{|c|c|c|c|c|c|c|}
\hline X \text { (количество спам-писем) } & 0 & 1 & 2 & 3 & 4 & 5 & 6 & 7 \\
\hline \mathrm{P} \text { (вероятность) } & 0.1 & 0.3 & 0.1 & 0.25 & 0.05 & 0.05 & 0.1 & 0.05 \\
\hline
\end{array}
$$

2. Найдем математическое ожидание для количества полученных писем.

Математическое ожидание случайной величины — это взвешенное среднее значений этой величины с учётом вероятностей. Оно вычисляется по следующей формуле:

$E(X) = \mu_X = \sum x_i \cdot P(X = x_i)$, где

$i$ - индекс переменной,

$x_i$ - значение случайной величины с индексом $i$,

$P(X = x_i)$ = вероятность того, что случайная величина принимает значение $x_i$.

$E(X) = x_0 * P(X = 0) + x_1 * P(X = 1) + x_2 * P(X = 2) + x_3 * P(X = 3) + x_4 * P(X = 4) + x_5 * P(X = 5) + x_6 * P(X = 6) + x_7 * P(X = 7) = 0*0.1 + 1*0.3 + 2*0.1 + 3*0.25 + 4*0.05 + 5*0.05 + 6*0.1 + 7*0.05 = 2.65$

$E(X) = 2.65$

3. Найдем дисперсию для количества полученных писем.
Дисперсия — это мера, которая показывает, насколько широко или узко разбросаны данные вокруг их среднего значения (математического ожидания). Она вычисляется по следующей формуле:

$V(\mathrm{X})=D(\mathrm{X})=\sigma_{X}^{2}=E\left(\mathrm{X}^{2}\right)-(E(\mathrm{X}))^{2}$,

Вычислим дисперсию для нашего случая:

$V(\mathrm{X})=D(\mathrm{X})=\sigma_{X}^{2} = E(X^2) - (E(X))^2 = ({x_0}^2 * P(X = 0) + {x_1}^2 * P(X = 1) + {x_2}^2 * P(X = 2) + {x_3}^2 * P(X = 3) + {x_4}^2 * P(X = 4) + {x_5}^2 * P(X = 5) + {x_6}^2 * P(X = 6) + {x_7}^2 * P(X = 7)) - (2.65)^2 = (0*0.1 + 1*0.3 + 4*0.1 + 9*0.25 + 16*0.05 + 25*0.05 + 36*0.1 + 49*0.05) - 7.0225 = 11.05 - 7.0225 = 4.0275 \approx 4.03$

Вычислим стандартное отклонение:

$\sigma_{X} = \sqrt{V(\mathrm{X})} = \sqrt{4.03} = 2.01$

$V(\mathrm{X}) = 4.03$

$\sigma_{X} = 2.01$

### **Ответ: вероятность получения 7 спам-писем - $0.05$, математическое ожидание для количества полученных писем - $E(X) = 2.65$, дисперсия для количества полученных писем - $V(\mathrm{X}) = 4.03$.**

### **Задание 7**
За 8 часов рабочего дня в колл-центр поступает в среднем 16 звонков. Найдите вероятность, что за час в колл-центр поступит не более 4 и не менее 2 звонков. Ответ округлите до тысячных. Задание должно быть решено без использования готовых функций Python.

Вычислим вероятность того, что за час в колл-центр поступит не более 4 и не менее 2 звонков.
Нам необходимо найти вероятность того, что поступит не более 4 и не менее 2 звонков $P(2 \geq X \leq 4)$. Это значит, что необходимо найти сумму вероятностей поступления каждого количества звонков из интервала $[2; 4]$.

В нашем случае есть событие (звонок в колл-центр), дана вероятность этого события за определенный период времени (8 часов), что соответсвует распределению Пуассона. (Распределение Пуассона — это дискретное распределение вероятностей числа событий, происходящих в данный период времени, с учётом среднего количества раз, когда событие происходит за этот период времени.)

В распределении Пуассона значение случайной величины может быть любым неотрицательным числом.  Случайная величина будет обладать следующими характеристиками:

$EX  = \lambda$ - математическое ожидание

$\sigma_X = \sqrt{\lambda}$ стандартное отклонение,

где  — $\lambda$ ожидаемое число событий за период времени.

$P_{call}(2 \geq X \leq 4) = P_{call}(X = 2) + P_{call}(X = 3) + P_{call}(X = 4)$


Чтобы рассчитать вероятность того, что за период времени произойдёт  $k$ событий, можно пользоваться следующей формулой:

$P(X = k) = \frac{\lambda^k e^{- \lambda}}{k!}$, где

$k$ - количество звонков, поочередно будет принимать значение 2,3,4

$\lambda =  16/8 = 2$ - ожидаемое количество звонков за 1 час

$P_{call}(X = 2) = \frac{2^2 * 2.71828^{- 2}}{2!} = \frac{4 * 0.13533546530402737}{2} = \frac{0.5413418612161095}{2} = 0.27067093060805475$

$P_{call}(X = 3) = \frac{2^3 * 2.71828^{- 2}}{3!} = \frac{8 * 0.13533546530402737}{6} = \frac{1.082683722432219}{6} = 0.1804472870720365$

$P_{call}(X = 4) = \frac{2^4 * 2.71828^{- 2}}{4!} = \frac{16 * 0.13533546530402737}{24} = \frac{2.165367444864438}{24} = 0.09022364353601825$

$P_{call}(2 \geq X \leq 4) = 0.27067093060805475 + 0.1804472870720365 + 0.09022364353601825 = 0.5413418612161095 \approx 0.541$

### **Ответ: вероятность того, что за час в колл-центр поступит не более 4 и не менее 2 звонков составляет $P_{call}(2 \geq X \leq 4) \approx 0.541$**